query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/mod.rs

```diff
@@ -284,52 +308,47 @@ impl fmt::Display for ConnectorVersion {
 /// Determines whether or not a test should run for the given enabled connectors and capabilities
 /// a connector is required to have.
 pub(crate) fn should_run(
+    connector: &ConnectorTag,
+    version: &ConnectorVersion,
     only: &[(&str, Option<&str>)],
     exclude: &[(&str, Option<&str>)],
     capabilities: ConnectorCapabilities,
 ) -> bool {
-    let (connector, version) = CONFIG.test_connector().unwrap();
-
     if !capabilities.is_empty() && !connector.capabilities().contains(capabilities) {
         println!("Connector excluded. Missing required capability.");
         return false;
     }
 
-    // We skip tests that exclude JS driver adapters when an external test executor is configured.
-    // A test that you only want to run with rust drivers can be annotated with exclude(JS)
-    if CONFIG.external_test_executor().is_some() && exclude.iter().any(|excl| excl.0.to_uppercase() == "JS") {
-        println!("Excluded test execution for JS driver adapters. Skipping test");
-        return false;
-    };
-    // we consume the JS token to prevent it from being used in the following checks
-    let exclude: Vec<_> = exclude.iter().filter(|excl| excl.0.to_uppercase() != "JS").collect();
-
-    // We only run tests that include JS driver adapters when an external test executor is configured.
-    // A test that you only want to run with js driver adapters can be annotated with only(JS)
-    if CONFIG.external_test_executor().is_none() && only.iter().any(|incl| incl.0.to_uppercase() == "JS") {
-        println!("Excluded test execution for rust driver adapters. Skipping test");
-        return false;
-    }
-    // we consume the JS token to prevent it from being used in the following checks
-    let only: Vec<_> = only.iter().filter(|incl| incl.0.to_uppercase() != "JS").collect();
+    let exclusions = exclude
+        .iter()
+        .filter_map(|c| ConnectorVersion::try_from(*c).ok())
+        .collect::<Vec<_>>();
+
+    let inclusions = only
+        .iter()
+        .filter_map(|c| ConnectorVersion::try_from(*c).ok())
+        .collect::<Vec<_>>();
 
-    if !only.is_empty() {
-        return only
-            .iter()
-            .any(|incl| ConnectorVersion::try_from(**incl).unwrap().matches_pattern(&version));
+    for exclusion in exclusions.iter() {
+        for inclusion in inclusions.iter() {
+            if exclusion.is_broader(inclusion) {
+                panic!("Error in connector test execution rules. `{exclusion}` in `excluded()` subsumes `{inclusion}` in `only()`");
```

Comment:
I would say, use simple English for this, avoid `subsumes` here to convey the meaning.